builtin/providers/azurerm/resource_arm_virtual_network.go

```diff
@@ -0,0 +1,256 @@
+				Type:     schema.TypeList,
+				Required: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+
+			"dns_servers": &schema.Schema{
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+
+			"subnet": &schema.Schema{
+				Type:     schema.TypeSet,
+				Required: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": &schema.Schema{
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"address_prefix": &schema.Schema{
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"security_group": &schema.Schema{
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+					},
+				},
+				Set: resourceAzureSubnetHash,
+			},
+
+			"location": &schema.Schema{
+				Type:      schema.TypeString,
+				Required:  true,
+				ForceNew:  true,
+				StateFunc: azureRMNormalizeLocation,
+			},
+
+			"resource_group_name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+		},
+	}
+}
+
+// resourceArmVirtualNetworkCreate creates the specified ARM virtual network.
+func resourceArmVirtualNetworkCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient)
+	vnetClient := client.vnetClient
+
+	log.Printf("[INFO] preparing arguments for Azure ARM virtual network creation.")
+
+	name := d.Get("name").(string)
+	location := d.Get("location").(string)
+	resGroup := d.Get("resource_group_name").(string)
+
+	vnet := network.VirtualNetwork{
+		Name:       &name,
+		Location:   &location,
+		Properties: getVirtualNetworkProperties(d),
+	}
+
+	log.Printf("[INFO] Sending virtual network create request to ARM.")
+	_, err := vnetClient.CreateOrUpdate(resGroup, name, vnet)
+	if err != nil {
+		return err
+	}
+
+	d.SetId(name)
+
+	// Wait for the resource group to become available
+	// TODO(jen20): Is there any need for this?
+	log.Printf("[DEBUG] Waiting for Virtual Network (%s) to become available", d.Id())
+	stateConf := &resource.StateChangeConf{
+		Pending: []string{"Accepted", "Updating"},
+		Target:  "Succeeded",
+		Refresh: virtualNetworkStateRefreshFunc(client, resGroup, name),
+		Timeout: 10 * time.Minute,
+	}
+	if _, err := stateConf.WaitForState(); err != nil {
+		return fmt.Errorf("Error waiting for Virtual Network (%s) to become available: %s", d.Id(), err)
+	}
+
+	return resourceArmVirtualNetworkRead(d, meta)
+}
+
+// resourceArmVirtualNetworkRead goes ahead and reads the state of the corresponding ARM virtual network.
+func resourceArmVirtualNetworkRead(d *schema.ResourceData, meta interface{}) error {
+	vnetClient := meta.(*ArmClient).vnetClient
+
+	name := d.Id()
+	resGroup := d.Get("resource_group_name").(string)
+
+	log.Printf("[INFO] Sending virtual network read request to ARM.")
+
+	resp, err := vnetClient.Get(resGroup, name)
+	if resp.StatusCode == http.StatusNotFound {
```

Comment:
The exact point I was raising concerning `resourceArmResourceGroupRead`. Either this logic is flawed from Terraform's perspective, or it must added in the resource group Read() as well...